src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,408 @@
+    -- (which is at least one because we are not in `S`).
+    ... = (ε/2) + ∑ k in Sᶜ, abs (f k/ₙ - f x) * bernstein n k x
+                              : by rw [bernstein.probability, mul_one]
+    ... ≤ (ε/2) + ∑ k in Sᶜ, (2 * ∥f∥) * bernstein n k x
+                              : add_le_add_left (finset.sum_le_sum
+                                  (λ k m, mul_le_mul_of_nonneg_right (w₃ _ _) bernstein_nonneg)) _
+    ... = (ε/2) + (2 * ∥f∥) * ∑ k in Sᶜ, bernstein n k x
+                              : by rw finset.mul_sum
+    ... ≤ (ε/2) + (2 * ∥f∥) * ∑ k in Sᶜ, δ^(-2 : ℤ) * (x - k/ₙ)^2 * bernstein n k x
+                              : add_le_add_left (mul_le_mul_of_nonneg_left
+                                  (finset.sum_le_sum (λ k m, begin
+                                    conv_lhs { rw ←one_mul (bernstein _ _ _), },
+                                    exact mul_le_mul_of_nonneg_right
+                                      (le_of_mem_S_compl m) bernstein_nonneg,
+                                  end)) w₁) _
+    -- Again enlarging the sum from `Sᶜ` to all of `fin (n+1)`
+    ... ≤ (ε/2) + (2 * ∥f∥) * ∑ k : fin (n+1), δ^(-2 : ℤ) * (x - k/ₙ)^2 * bernstein n k x
+                              : add_le_add_left (mul_le_mul_of_nonneg_left
+                                  (finset.sum_le_univ_sum_of_nonneg
+                                    (λ k, mul_nonneg
+                                      (mul_nonneg (pow_minus_two_nonneg _) (pow_two_nonneg _))
+                                      bernstein_nonneg)) w₁) _
+    ... = (ε/2) + (2 * ∥f∥) * δ^(-2 : ℤ) * ∑ k : fin (n+1), (x - k/ₙ)^2 * bernstein n k x
+                              : by conv_rhs {
+                                  rw [mul_assoc, finset.mul_sum], simp only [←mul_assoc], }
+    -- `bernstein.variance` and `x ∈ [0,1]` gives the uniform bound
+    ... = (ε/2) + (2 * ∥f∥) * δ^(-2 : ℤ) * x * (1-x) / n
+                              : by { rw variance npos, ring, }
+    ... ≤ (ε/2) + (2 * ∥f∥) * δ^(-2 : ℤ) / n
+                              : add_le_add_left ((div_le_div_right npos).mpr
+                                begin
+                                  apply mul_unit_interval_le w₂,
+                                  apply mul_unit_interval_le w₂ (le_refl _),
+                                  all_goals { unit_interval, },
+                                end) _
+    ... < ε : _, -- We postpone this final step for a moment, in order to actually choose `n`!
+  -- Choose `n` to make the inequality work.
+  show ℕ, { exact nat_ceil (2 * (2 * ∥f∥ * δ^(-2 : ℤ)) / ε) + 1, },
+  { -- And a final inequality bash gets us to the end.
+    dsimp [n] at npos ⊢,
+    rw [show ∀ z, ε/2 + z < ε ↔ z < ε/2, from λ z, by fsplit; { intro, linarith, }],
+    rw [lt_div_iff (show (0 : ℝ) < 2, by norm_num), mul_comm],
+    rw [←mul_div_assoc, div_lt_iff npos, mul_comm ε, ←div_lt_iff h],
+    exact lt_of_le_of_lt (le_nat_ceil _) (lt_add_one _), },
```

Comment:
This can be filter-y, rather than by choosing an explicit `n`.  Something like
```
  have h₁ : tendsto (λ n : ℕ, ε / 2 + 2 * ∥f∥ * δ ^ (-2 : ℤ) / n) at_top (𝓝 (ε / 2)),
  { sorry },
  have h₂ : ∀ᶠ y in 𝓝 (ε / 2), y < ε := gt_mem_nhds (half_lt_self h),
  obtain ⟨n, hn⟩ := (h₁.eventually h₂).exists,
```
I think `tendsto.const_add` and `tendsto_const_div_at_top_nhds_0_nat` are all that's needed for the sorry.

Reply:
In fact, why not make the statement of your theorem a "tendsto"?
```lean
theorem bernstein_approximation_uniform (f : I →ᵇ ℝ) :
  tendsto (λ n : ℕ, bernstein_approximation n f) at_top (𝓝 f) :=
```
The proof will be almost exactly the same.

Reply:
Okay, I've made the whole theorem a `tendsto`.

Reply:
That then means I'm not sure how to follow your first ("filter-y") suggestion. At the end of the proof I now have two goals:
`2 * ∥f∥ * δ ^ -2 / ↑n < ε / 2`, knowing that `N < n` for some unspecified `N`, and `ℕ`, i.e. that `N`.

My current strategy works fine: call `show ℕ` to switch goals, pick a concrete value, then deal with the inequality. I'm not sure how to use the filter API, however.